lib/logstash/outputs/redis.rb

```diff
@@ -92,6 +92,9 @@ class LogStash::Outputs::Redis < LogStash::Outputs::Base
   # Only supported for `list` Redis `data_type`.
   config :congestion_threshold, :validate => :number, :default => 0
 
+	# Auto-Queue switching when full, instead of waiting.
+  config :congestion_switch, :validate => :boolean, :default => false
```

Comment:
This should be the switch to determine if the switch should be done or not. (eg if this patch should be enabled or if old function should be used).